src/optimization/utils/variables.jl

```diff
@@ -22,13 +22,13 @@ function init_variables(model::Model, layers::Vector{Layer}; binary = false, inc
     end
 
     for (i, n) in enumerate(all_layers_n)
-        vars[i] = @variable(model, [1:n], binary = binary, base_name = "z$i")
+        vars[i] = @variable(model, [1:n], binary = binary, base_name = string(name, "$i"))
     end
     return vars
 end
 
 function symbolic_max(m::Model, a, b)
-    aux = @variable(m)
+    aux = @variable(m, base_name = "aux_max")
```

Comment:
Unfortunately, we can't do this `base_name = "aux_max"` business so easily here. `symbolic_max` (and the others) will be called more than once per model and a name shouldn't ("can't") be shared between multiple variables. If we really want these to be named we can come up with a scheme to name them sequentially, but it might be a slight hassle.